### PR TITLE
[docs] Add AGENTS.md and CLAUDE.md for AI coding agent context

### LINKED ISSUE #1078: [Feature] Add AGENTS.md for AI coding agent context
### Search before asking

- [x] I had searched in the [issues](https://github.com/project-codeflare/codeflare-sdk/issues) and found no similar feature requirement.

### Description

I'd like to propose adding an `AGENTS.md` file to the repository root, along with a `CLAUDE.md` symlink pointing to it. This gives AI coding agents (Claude Code, Codex, GitHub Copilot, etc.) repository context so they can work more effectively.

The repo already has excellent `.cursor/rules/` files for Cursor users. This change extends that coverage to other AI coding tools by providing a condensed context file documenting:

* **Repository structure**: The `src/codeflare_sdk/` layout (`common/`, `ray/`, `vendored/`) and key packages.

* **Setup and test commands**: Poetry install, pytest with coverage, pre-commit commands.

* **Single-file commands**: How to format individual files (`black path/to/file.py`).

* **Coverage requirements**: >=90% project, >=85% patch.

* **Coding conventions**: Import ordering, type hints, Google-style docstrings, Apache-2.0 header, naming conventions.

* **Kubernetes API patterns**: `config_check()`, `get_api_client()`, `_kube_api_error_handling()` usage.

* **Testing patterns**: pytest-mock, test helpers from `unit_test_support.py`, edge-case testing guidance.

Using `AGENTS.md` as the source of truth (cross-tool convention) with a `CLAUDE.md` symlink keeps a single file to maintain while ensuring compatibility with Claude Code, which auto-discovers `CLAUDE.md`.

This is a docs-only change — no code is modified. The file is 142 lines, kept concise and complementary to the existing `.cursor/rules/`.

### Use case

AI coding agents are increasingly used for development tasks across open source projects. Without a context file, agents must rediscover the repo layout, build commands, and conventions from scratch each session — leading to slower iteration, incorrect commands, and style violations. The existing `.cursor/rules/` files solve this for Cursor but not for other tools. A lightweight `AGENTS.md` eliminates this gap.

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by abiazett**
### Summary

Add `AGENTS.md` (cross-tool AI agent context file) and a `CLAUDE.md` symlink
pointing to it, giving AI coding agents repo context: project layout,
build/test commands, coding conventions, and key patterns.

### Why

AI coding agents (Claude Code, Codex, GitHub Copilot, etc.) work more
effectively when they have a context file documenting repo structure and
commands. The repo already has excellent `.cursor/rules/` for Cursor users —
this extends coverage to other AI coding tools.

- `AGENTS.md` is the cross-tool convention (Codex, GitHub Copilot, etc.)
- `CLAUDE.md` symlink ensures Claude Code auto-discovers it
- Single source of truth, no content duplication
- Complements existing `.cursor/rules/` (condensed, not duplicated)

### What's included

The file documents (142 lines):

- **Repository structure**: `src/codeflare_sdk/` layout with `common/`, `ray/`, `vendored/`
- **Setup**: Poetry install commands
- **Build and test commands**: pytest with coverage, pre-commit
- **Single-file commands**: `black path/to/file.py`
- **Coverage requirements**: >=90% project, >=85% patch
- **Coding conventions**: import order, type hints, Google-style docstrings, Apache-2.0 header
- **K8s API patterns**: `config_check()`, `get_api_client()`, `_kube_api_error_handling()`
- **Testing patterns**: pytest-mock, `unit_test_support.py` helpers, edge-case guidance

### Testing

- `pre-commit run --files AGENTS.md CLAUDE.md` passes
- No code changes — all existing CI should pass unchanged

Closes #1078

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+```sh
+# Pre-commit (formatting + checks)
+pre-commit run --show-diff-on-failure --color=always --all-files
+
+# Unit tests with coverage (excludes E2E, notebooks, vendored)
+coverage run \
+  --omit="src/**/test_*.py,src/codeflare_sdk/common/utils/unit_test_support.py,src/codeflare_sdk/vendored/**" \
+  -m pytest \
+  --ignore=tests/e2e --ignore=tests/e2e_v2 --ignore=tests/upgrade \
+  --ignore=demo-notebooks --ignore=ui-tests
```
Comment (pawelpaszki) [inline AGENTS.md]:
```suggestion
  --ignore=demo-notebooks --ignore=tests/ui
```

**Comment by pawelpaszki** [comment]
Thanks Ana! minor comment added. the rest looks good

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+### Pre-Commit Hooks
+
+Pre-commit hooks enforce:
+
+- trailing-whitespace removal
+- end-of-file newline
+- YAML validation
+- Large file checks
+- black formatting
```
Comment (chipspeak) [inline AGENTS.md:142]: I'd add a section that points to our cursor rules for the benefit of agents that can read files. @DavidAdaRH implemented these and they've demonstrated improvements in output so I think it's no harm to point to them.

It could just be added to the end of the file like:

`## Cursor Rules (extended guidance)
This repository has more detailed AI coding rules in `.cursor/rules/`:
- `.cursor/rules/01-project-context.mdc` — Grounding, personas, hallucination avoidance
- `.cursor/rules/02-python-standards.mdc` — Python style, canonical examples, common pitfalls
- `.cursor/rules/03-testing-and-ci.mdc` — CI workflows, demo notebooks, KinD adaptations`
